core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1464,9 +1464,13 @@ class Log(@volatile var dir: File,
       var segmentEntry = segments.floorEntry(startOffset)
 
       // return error on attempt to read beyond the log end offset or read below log start offset
-      if (startOffset > endOffset || segmentEntry == null || startOffset < logStartOffset)
-        throw new OffsetOutOfRangeException(s"Received request for offset $startOffset for partition $topicPartition, " +
-          s"but we only have log segments in the range $logStartOffset to $endOffset.")
+      if (startOffset > endOffset || segmentEntry == null || startOffset < logStartOffset) {
+        val msg = s"Received request for offset $startOffset for partition $topicPartition, " +
+          s"but we only have log segments in the range $logStartOffset to $endOffset."
+        assert(logStartOffset == 0, "logStartOffset = " + logStartOffset)
```

Comment:
I don't get this assert. Why do we want logStartOffset to be 0?

Reply:
Misunderstood. Deleted.